src/physics/em/EnergyLossDistribution.i.hh

```diff
@@ -0,0 +1,362 @@
+ */
+template<class Engine>
+CELER_FUNCTION auto EnergyLossDistribution::operator()(Engine& rng) const
+    -> MevEnergy
+{
+    // Small step or low density material
+    if (mean_loss_ < EnergyLossDistribution::min_valid_energy().value()
+        || max_energy_ <= EnergyLossDistribution::ionization_energy().value())
+    {
+        return MevEnergy{mean_loss_};
+    }
+
+    // The Gaussian approximation is valid for heavy particles and in the
+    // regime \f$ \kappa > 10 \f$. Fluctuations of the unrestricted energy loss
+    // follow a Gaussian distribution if \f$ \Delta E > \kappa T_{max} \f$,
+    // where \f$ T_{max} \f$ is the maximum energy transfer (PHYS332 section
+    // 2). For fluctuations of the \em restricted energy loss, the condition is
+    // modified to \f$ \Delta E > \kappa T_{c} \f$ and \f$ T_{max} \le 2 T_c
+    // \f$, where \f$ T_c \f$ is the delta ray cutoff energy (PRM Eq. 7.6-7.7).
+    if (mass_ratio_ < 1
+        && mean_loss_ >= EnergyLossDistribution::min_kappa() * max_energy_
+        && max_energy_transfer_ <= 2 * max_energy_)
+    {
+        // Approximate straggling function as a Gaussian distribution
+        return MevEnergy{this->sample_gaussian(rng)};
+    }
+    // Use Urban model of energy loss fluctuations in thin layers
+    return MevEnergy{this->sample_urban(rng)};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Gaussian model of energy loss fluctuations.
+ *
+ * In a thick absorber, the total energy transfer is a result of many small
+ * energy losses from a large number of collisions. The central limit theorem
+ * applies, and the energy loss fluctuations can be described by a Gaussian
+ * distribution. See section 7.3.1 of the Geant4 Physics Reference Manual and
+ * GEANT3 PHYS332 section 2.3.
+ */
+template<class Engine>
+CELER_FUNCTION real_type EnergyLossDistribution::sample_gaussian(Engine& rng) const
+{
+    // Square root of Bohr's variance (PRM Eq. 7.8). For thick absorbers, the
+    // straggling function approaches a Gaussian distribution with this
+    // standard deviation
+    const real_type stddev = std::sqrt(
+        2 * constants::pi * ipow<2>(constants::r_electron)
+        * shared_.electron_mass * material_.material_view().electron_density()
+        * charge_sq_ * max_energy_ * step_length_
+        * (1 / beta_sq_ - real_type(0.5)));
+
+    // Sample energy loss from a Gaussian distribution
+    if (mean_loss_ >= 2 * stddev)
+    {
+        real_type                     result;
+        const real_type               max_loss = 2 * mean_loss_;
+        NormalDistribution<real_type> sample_normal(mean_loss_, stddev);
+        do
+        {
+            result = sample_normal(rng);
+        } while (result <= 0 || result > max_loss);
+        return result;
+    }
+    // Sample energy loss from a gamma distribution
+    const real_type k = ipow<2>(mean_loss_ / stddev);
+    return GammaDistribution<real_type>(k, k)(rng);
```

Comment:
ohhh good catch! I totally missed the `meanLoss` in front. In addition to being confused about why the loss is sampled from a gamma distribution here in the first place, I didn't understand why the distribution mean was not the mean loss -- this makes a little more sense now. Looking at this again, maybe CLHEP *is* using the same parameterization for the gamma distribution, and this should be `GammaDistribution<real_type>(k, mean_loss_ / k)(rng)`?

Reply:
I honestly couldn't figure out which of the parameterizations CLHEP uses. There don't seem to be any comments in the code that indicate it. Apparently the implementation comes from [C-rand](https://www.researchgate.net/publication/239650613_C-rand_a_package_for_generating_nonuniform_random_variates), but the documentation for C-rand doesn't say explicitly either. I think the best thing to do would be to use our version and the CLHEP version to generate a bunch of variates and bin them in a histogram to see how they look compared to one another.

Reply:
I guess it doesn't actually matter here which they use since they multiply by the scale parameter after sampling, but for reference CLHEP does use the inverse scale parameterization:
![gamma](https://user-images.githubusercontent.com/6426426/129791989-700dbb84-57bf-4169-ab29-1d522a05921d.png)
Anyway, thanks! Fixed this and updated the test.